services/coverity/coverity-scan.tester.js

```diff
@@ -0,0 +1,71 @@
+'use strict'
+
+const Joi = require('joi')
+const t = (module.exports = require('../create-service-tester')())
+
+t.create('live: known project id')
+  .get('/3997.json')
+  .expectJSONTypes(
+    Joi.object().keys({
+      name: 'coverity',
+      value: Joi.string().regex(/passing|passed .* new defects|pending|failed/),
```

Comment:
Lets move `messageRegex` out into a helper module so we can use the same regex in the test here and the validation schema.

Reply:
I almost did that but there is a _very_ subtle difference in the regex. Coverity returns `passed` (and variances of passed), but the LegacyService implementation was converting that to `passing`. I'm not sure why `passed` was converted to `passing` but the other status (`passed 5 new defects`) kept `passed` in the message.

Reply:
Well spotted. Looks like that got done here: https://github.com/badges/shields/pull/585

I don't have strong opinions on it. Happy for you to either leave it as-is. I only mentioned it because I didn't notice that the 2 regexes aren't exactly the same.